bundles/org.pitest.pitclipse.ui/src/org/pitest/pitclipse/ui/view/PitView.java

```diff
@@ -53,9 +96,57 @@ public void setFocus() {
     @Override
     public synchronized void update(File result) {
         if (result == null) {
+            homeUrlString = null;
             browser.setText("<html/>");
         } else {
-            browser.setUrl(result.toURI().toString());
+            homeUrlString = result.toURI().toString();
+            // fallback, if the file does not exist
+            browser.setUrl(homeUrlString);
+            String url = getUrlToLastActiveClass(result);
+            if (url != null && new File(url).exists()) {
+                browser.setUrl(url);
+            }
+        }
+    }
+
+    /**
+     * Looks up the last active file from an editor and constructs a path to its
+     * page of the result
+     * 
+     * @param result file which is used for the path beginning
+     * @return String which represents the path in the result folder or null, if the
+     *         file form the editor is no .java file or no FileEditor was active
+     *         last
+     */
+    private String getUrlToLastActiveClass(File result) {
+        IEditorPart editor = getSite().getPage().getActiveEditor();
```

Comment:
I'm wondering if you could do something along the lines of `org.pitest.pitclipse.launch.ui.PitLaunchShortcut.getLaunchableResource(IEditorPart)`, which also relies on `org.eclipse.jdt.ui.JavaUI.getEditorInputTypeRoot(IEditorInput)`